Automattic/jetpack
# Update/inform first connected plugin

When the `ConnectButton` uses the `api` js-package's `registerSite` method, it makes a call to `v4/register` without any identification of which is the plugin making that request.

The registration is done with a generic instance of the Connection Manager and, as consequence, the `connect_plugin` argument is not sent to the registration endpoint in WPCOM. 

The result is that we don't have the information of which plugin performed the registration.

This PR introduces the usage of the new Config JS package to identify which plugin is consuming the API RNA package and adds a new parameter to the registration endpoint.

#### Changes proposed in this Pull Request:

* Adds a new parameter to the `v4/register` endpoint to inform the first plugin to register the site.

#### Jetpack product discussion


1191179647901802-as-1201167654915231

#### Does this pull request change what data or activity we track or use?


No, it fixes some cases in which information we already track was not being properly sent

#### Testing instructions:






* Set up a new site with only Jetpack the plugin and connect it
* In your sandbox, verify that the `jetpack_first_connected_plugin` was added to the cache site with the correct slug (in this case, `jetpack`)
* Repeat the above with only the Backup plugin active, and Boost, and all combinations you can come up with. Make sure the value of the `jetpack_first_connected_plugin` option always match the slug of the plugin you used to establish the connection.

**Question**: I'm checking to see if the property exists in webpack.config to avoid JS throwing an error, but my preference is to let the package throw to make sure this is a required config and ensure future consumers to always inform it. However, technically, this introduces a breaking change and requires a major version bump of the API js package. What do you think?
